pxr/usd/sdf/textFileFormat.ll

```diff
@@ -134,25 +164,93 @@ PXR_NAMESPACE_USING_DIRECTIVE
 "variants"            { (*yylval_param) = std::string(yytext, yyleng); return TOK_VARIANTS; }
 "varying"             { (*yylval_param) = std::string(yytext, yyleng); return TOK_VARYING; }
 
-    /* unquoted C/Python identifier */
-[[:alpha:]_][[:alnum:]_]* {
-        (*yylval_param) = std::string(yytext, yyleng);
-        return TOK_IDENTIFIER;
+ /* In a Unicode enabled scheme, 'identifiers' are generally
+  * categorized as something that begins with something in the
+  * XID_Start category followed by zero or more things in the
+  * XID_Continue category.  Since the number of characters in
+  * these classes are large, we can't explicitly validate them
+  * here easily, so the lex rule is pretty permissive with some
+  * further validation done in code prior to calling what was
+  * read an 'identifier'.  Note this rule will also match
+  * standard ASCII strings because the UTF-8 encoded byte 
+  * representation is the same for these characters.
+  * However, unlike the path lexer, we can guarantee that 
+  * prim names aren't something special to be called out here
+  * so we can be a little more specific about the kinds of strings
+  * we match, particularly to not collide with the pure digit match rule
+  * below
+  */
+{UTF8NODIGU}{UTF8U}* {
+    std::string matched = std::string(yytext, yyleng);
+
+    // we perform an extra validation step here
+    // to make sure what we matched is actually a valid
+    // identifier because we can overmatch UTF-8 characters
+    // based on this definition
+    if (!SdfPath::IsValidIdentifier(matched)) {
```

Comment:
Hmm -- I think we need a non-SdfPath specific function that checks if an identifier is the utf-8 style.  This TOK_IDENTIFIER thing is used for other things in the parser like dictionary keys, or metadata keys, not just prim & property names.  Which raises the question, are we intending to add utf-8 support for those elements too?  Or would it be preferable to scope this just to prim & property names?